content/articles/naive-bayes-algorithm-in-python/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /naive-bayes-algorithm-in-python/
+title: A Step by Step Guide to Implement Naive Bayes Algorithm in R
+description: This tutorial will discuss the Naive Bayes algorithm, the real-world application of the Naive Bayes algorithm, implement and evaluate its performance using a confusion matrix in R.
+author: lawrence-mbici
+date: 2021-11-11T00:00:00-21:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/naive-bayes-algorithm-in-python/hero.png 
+    alt: A Step by Step Guide to Implement Naive Bayes Algorithm in R
+---
+Naive Bayes is a machine learning algorithm based on the Bayes Theorem, and it's used for classification problems. The naive Bayes classifier is very effective and can be used with highly complex problems despite its simplicity. Due to its ability to handle highly complex tasks, the Naive Bayes has gained popularity in machine learning for a long time. Some Naive Bayes applications include; sentiment analysis, spam filtering, text classification, and many more.
+<!--more-->
+This tutorial will discuss the Naive Bayes algorithm and its principles to deliver a solid and clear understanding of this tool. Later, we will discuss the real-world application of the Naive Bayes algorithm and finally implement and evaluate its performance using a confusion matrix in R.
+
+### Prerequisite
+To follow along with this material comfortably, you're required to have:
+- [R](https://cran.r-project.org/) installed on your computer
+- Programming skills in R
+- A [dataset](https://github.com/mbici/data/blob/main/data.csv) that we shall use in our implementation
+- Require packages installed, i.e., 
+  1. install.packages('caTools')
+  2. install.packages('e1071')
+
+### Introduction to the Bayes Algorithm
+Before diving into *Naive Bayes*, we must first understand the *Bayes Theorem* and its assumptions. To understand this, first, we consider a conditional probability from which the mathematical representation of the *Bayes theorem* is derived. From the probability and statistic world, the conditional probability is defined as:
+
+$p(A|B)=\frac{p(A\cap B)}{p(B)}$
+
+Where:
+- A and B are two events.
+- $p(A|B)$ is the conditional probability. We read this as the probability of *A* given *B*.
+- $p(A\cap B)=p(A, B)$  is called the join probability, i.e.,  probability of A and B happening together.
```

Comment:
This should be `joint probability`. Not `join probability`. Please edit accordingly